File: frontend/src/types/product.ts
export interface Product {
  code: string;
  name: string;
  nutriments: {
    carbohydrates: number;
    proteins: number;
    fat: number;
    "energy-kcal": number;
  };
}

export interface CartResponse {
  total_items: number;
  products: Product[];
}

File: app/utils/product.py
class Product:
    def __init__(self, code, name, nutriments):
        self._code = code
        self._name = name
        self._nutriments = nutriments

    @property
    def code(self):
        return self._code

    @property
    def name(self):
        return self._name
    
    @property
    def nutriments(self):
        return self._nutriments